worker/peergrouper/mock_test.go

```diff
@@ -125,8 +123,7 @@ func (st *fakeState) checkInvariants() {
 	if err := st.check(st); err != nil {
 		// Force a panic, otherwise we can deadlock
 		// when called from within the worker.
-		go panic(err)
-		select {}
```

Comment:
I believe that this was deliberate, so that if invariants are broken you will see stack traces of all goroutines rather than just the one we called panic in.